solidity-v1/dashboard/src/pages/threshold/ThresholdUpgradePage.jsx

```diff
@@ -0,0 +1,316 @@
+import React, { useEffect, useMemo } from "react"
+import * as Icons from "../../components/Icons"
+import NavLink from "../../components/NavLink"
+import TokenAmount from "../../components/TokenAmount"
+import { KEEP } from "../../utils/token.utils"
+import AllocationProgressBar from "../../components/threshold/AllocationProgressBar"
+import UpgradeTokensTile from "../../components/threshold/UpgradeTokensTile"
+import resourceTooltipProps from "../../constants/tooltips"
+import useKeepBalanceInfo from "../../hooks/useKeepBalanceInfo"
+import useGrantedBalanceInfo from "../../hooks/useGrantedBalanceInfo"
+import { add, lte } from "../../utils/arithmetics.utils"
+import {
+  useWeb3Address,
+  useWeb3Context,
+} from "../../components/WithWeb3Context"
+import { useDispatch, useSelector } from "react-redux"
+import { useModal } from "../../hooks/useModal"
+import { LINK, MODAL_TYPES } from "../../constants/constants"
+import TokenAmountSkeleton from "../../components/skeletons/TokenAmountSkeleton"
+import ResourceTooltip from "../../components/ResourceTooltip"
+
+const ThresholdUpgradePage = () => {
+  const { isConnected } = useWeb3Context()
+  const address = useWeb3Address()
+  const dispatch = useDispatch()
+  const { openModal } = useModal()
+
+  useEffect(() => {
+    if (isConnected) {
+      dispatch({
+        type: "staking/fetch_delegations_request",
+        payload: { address },
+      })
+      dispatch({
+        type: "token-grant/fetch_grants_request",
+        payload: { address },
+      })
+    }
+  }, [dispatch, isConnected, address])
+
+  const { delegations, undelegations, isDelegationDataFetching } = useSelector(
+    (state) => state.staking
+  )
+
+  const { isFetching: isGrantDataFetching } = useSelector(
+    (state) => state.tokenGrants
+  )
+
+  const isDataFetching = useMemo(() => {
+    return isDelegationDataFetching || isGrantDataFetching
+  }, [isDelegationDataFetching, isGrantDataFetching])
+
+  const { totalOwnedStakedBalance, totalOwnedUnstakedBalance } =
+    useKeepBalanceInfo()
+
+  const { totalGrantedStakedBalance, totalGrantedUnstakedBalance } =
+    useGrantedBalanceInfo()
+
+  const notStakedTotalAmount = useMemo(() => {
+    return add(
+      totalOwnedUnstakedBalance,
+      totalGrantedUnstakedBalance
```

Comment:
`totalGrantedUnstakedBalance ` doesn't mean the grantee can withdraw this amount. I believe we should only take into account tokens that are ready to release. `grant.readyToRelease`.

Reply:
Fixed in 605b9f60dfafd120bbe8624967fe496e9a9d9f37